src/cupbearer/detectors/anomaly_detector.py

```diff
@@ -133,13 +133,15 @@ def eval(
             test_loader = tqdm(test_loader, desc="Evaluating", leave=False)
         for batch in test_loader:
             inputs, new_labels = batch
-            new_scores = self.layerwise_scores(inputs)
-            scores.append(new_scores.mean(axis=0))
-            # Sum over batch axis
-            layer_scores = layer_scores + new_scores.sum(axis=1)
-            num_elements += new_scores.shape[1]
+            try:
+                new_scores = self.layerwise_scores(inputs)
+                scores.append(new_scores.mean(axis=0))
```

Comment:
I don't think I follow---the intended behavior is that if `scores()` is overriden and layerwise_scores rasises a NotImplementedError, we use the new `scores` method in the child class. Are you saying that isn't what will happen, or that there's something wrong with that?

If we add a final decorator inside `AnomalyDetector`, that would currently break the new finetuning detector since it overrides `scores()` (because it can't implement scoring layerwise). But maybe I'm misunderstanding your suggestion?

Reply:
In cases where `scores()` is overridden and `layerwise_scores()` does not raise `NotImplementedError` you might still expect scores to be calculated by `scores()` instead of `layerwise_scores(inputs).mean(axis=0)`.

But, yes I didn't think this through.

We could still add check to see if scores are overridden (something like `type(self).scores == AnomalyDetector.scores`) but that is a lot less clean, so we could just skip this. Though, a comment that `scores()` are unused if `layerwise_scores()` is implemented would be nice.

Reply:
Oh, I see what you mean. Yeah, overriding `scores` while also implementing `layerwise_scores` is something I didn't intend to support atm. I actually think your suggested check is best here---it does feel a bit strange, but that's probably the fault of the overall design. If we have a check at least things won't fail quietly.